core/common/src/RpcInterface.ts

```diff
@@ -54,14 +59,117 @@ export abstract class RpcInterface {
+      const request = new (this.configuration.protocol.requestType as any)(this, operationName, parametersArray) as RpcRequest;
+      request.submit(); // eslint-disable-line @typescript-eslint/no-floating-promises
+      (this as any)[CURRENT_REQUEST] = request;
+      return request.response;
+    }
   }
 
   /** @internal */
   public configurationSupplier: RpcConfigurationSupplier | undefined;
 }
 
 RpcInterface.prototype.configurationSupplier = undefined;
+
+class InterceptedRequest extends RpcRequest {
+  protected override async load(): Promise<RpcSerializedValue> { throw new Error(); }
+  protected override async send(): Promise<number> { throw new Error(); }
+  protected override setHeader(_name: string, _value: string): void { throw new Error(); }
+}
+
+async function intercept(session: IpcSession, client: RpcInterface, operation: string, parameters: any[]) {
+  const request = new InterceptedRequest(client, operation, []);
+  (client as any)[CURRENT_REQUEST] = request;
+
+  const context = await client.configuration.protocol.serialize(request);
+  request.parameters = parameters;
+
+  const info: RPC = {
+    definition: {
+      interfaceName: context.operation.interfaceDefinition,
+      interfaceVersion: context.operation.interfaceVersion,
+    },
+    operation,
+    parameters,
+    context: {
+      applicationId: context.applicationId,
+      applicationVersion: context.applicationVersion,
+      id: context.id,
+      sessionId: context.sessionId,
+      protocolVersion: (context.protocolVersion || 0).toString(),
+    },
+  };
+
+  const dispatch = async () => {
+    aggregateLoad.lastRequest = new Date().getTime();
+
+    const response = await session.handleRpc(info);
+
+    aggregateLoad.lastResponse = new Date().getTime();
+
+    if (typeof (response) === "object" && response.hasOwnProperty("iTwinRpcCoreResponse") && response.hasOwnProperty("managedStatus")) {
+      const status: RpcManagedStatus = response;
+
+      if (status.managedStatus === "pending") {
+        return handlePending(request, status, dispatch);
+      } else if (status.managedStatus === "notFound") {
+        return handleNotFound(request, status, dispatch);
+      } else if (status.managedStatus === "noContent") {
+        return handleNoContent();
+      }
+    } else {
+      return response;
+    }
+  };
+
+  return dispatch();
+}
+
+async function handlePending(request: InterceptedRequest, status: RpcManagedStatus, dispatch: () => Promise<any>) {
+  request._status = RpcRequestStatus.Pending;
+  request._extendedStatus = status.responseValue;
+  RpcRequest.events.raiseEvent(RpcRequestEvent.PendingUpdateReceived, request);
+
+  const delay = request.operation.policy.retryInterval(request.client.configuration);
+
+  return new Promise((resolve, reject) => {
+    setTimeout(async () => {
+      try {
+        const response = await dispatch();
+        resolve(response);
+      } catch (err) {
+        reject(err);
+      }
+    }, delay);
+  });
```

Comment:
I think this can all be done without constructing a `Promise` or a try/catch...

```suggestion
  await BeDuration.wait(delay);
  return dispatch();
```

Reply:
Nice. Done...